src/plots/cartesian/axes.js

```diff
@@ -846,7 +847,27 @@ axes.tickIncrement = function(x, dtick, axrev, calendar) {
     var axSign = axrev ? -1 : 1;
 
     // includes linear, all dates smaller than month, and pure 10^n in log
-    if(isNumeric(dtick)) return x + axSign * dtick;
+    if(isNumeric(dtick)) {
+        // Note 1:
+        // 0.3 != 0.1 + 0.2 but 0.3 == ((10 * 0.1) + (10 * 0.2)) / 10
+        // Attempt to use integer steps to increment
+        var magic = 1 / dtick;
```

Comment:
Do you have a reference explaining that "magic"?

Reply:
Unfortunately no. But the trick appear to be working.

Reply:
Hmm, do we really need this block for `tickformat` values other than `'p'`?

Reply:
Yes it is also required e.g. with `s`.
Please find the new mock added in https://github.com/plotly/plotly.js/pull/4070/commits/740574b301991c39b2f868355565b639ab1e0b38 which currently renders as:
![before4064b](https://user-images.githubusercontent.com/33888540/61678776-d862b280-acd1-11e9-9e2d-186cce539ddb.png)

Reply:
> Yes it is also required e.g. with `s`.

Ok so could then have 

```js
if(isNumeric(dtick) && (tickformat.indexOf('s') !== -1 || tickformat.indexOf('p') !== -1)) {
  // your new logic
}
```

that would make this patch a lot less stressful to merge.

Reply:
@antoinerg @etpinard the logic is now moved to `Lib` in https://github.com/plotly/plotly.js/pull/4070/commits/916bf80e8d26d67c474bda6547c1ef4e1e53fe50 and I improved test coverage in https://github.com/plotly/plotly.js/pull/4070/commits/86b135dccaf5848338aaa1d909e9b48c2b3fa156.